### ISSUE TITLE
docs: $db->getFieldData() data explanation is not complete

### ISSUE THREAD

**Issue by kenjis** [codeigniter4/CodeIgniter4#6308]
`$db->getFieldData()`'s explanation is not complete and accurate.

```
The following data is available from this function if supported by your database:

    name - column name
    max_length - maximum length of the column
    primary_key - 1 if the column is a primary key
    type - the type of the column
```

https://codeigniter4.github.io/CodeIgniter4/database/metadata.html#db-getfielddata

**Comment by naente** [comment]
> not complete

I am considering adding nullable and default because they are not listed in the data that can be retrieved.
Is my understanding correct?

> accurate

Inaccurate means, for example
Postgre does not return primary key information... or something like that?

**Comment by kenjis** [comment]
Yes, something like that.
 I'm not sure Postgre does not return primary key.

My memo.
```diff
 -  name - column name
--  max_length - maximum length of the column
--  primary_key - 1 if the column is a primary key
 -  type - the type of the column
+-  type_name - the type name of the column
+-  max_length - maximum length of the column
+-  default - the default value
+-  primary_key - ``true`` if the column is a primary key
+-  nullable - ``true`` if the column is nullable
```

Anyway, the description is not accurate now, so any fixes are welcome.
If you send PRs, please show the actual output as well.

**Comment by naente** [comment]
Thank you.
Since it is difficult to prepare SQLite3, OCI8, and SQLSRV, all environments, I will update this page based on the results with MySQL.

Regarding the presentation of the actual output.
Is the attached screenshot sufficient?

`SHOW COLUMNS FROM items;`

<img width="960" src="https://user-images.githubusercontent.com/47913993/185752755-6738a2b8-6502-45b0-9c83-022f3dd55263.png">

`dd($db->getFieldData('items'));`

<img width="500" src="https://user-images.githubusercontent.com/47913993/185752837-4b35e357-3e28-47fe-b75c-97a8e3516cdf.png">

**Comment by kenjis** [comment]
No problem. MySQL is de fact standard of CI4. The results should be described first.
Also the screenshots are sufficient.

**Comment by naente** [comment]
I sent a PR: https://github.com/codeigniter4/CodeIgniter4/pull/6398

**Comment by kenjis** [comment]
I updated the test for `getFieldData()`: #6400 
And found that output results vary depending on the database. And some of them may be bugs.

I think it is better to write the specifications that should be in the user guide, because it is not very meaningful to describe the current behavior in detail.

It is not easy to decide what the specification should be, though.

**Comment by naente** [comment]
We checked the contents of the _fieldData method in Connection.php for each database under Codeigniter4/system/Database.

| | MySQLi | OCI8 | Postgre | SQLite3 | SQLSRV |
| --- | :---: | :---: | :---: | :---: | :---: |
| name | ● | ● | ● | ● | ● |
| type | ● | ● | ● | ● | ● |
| max_length | ● | ● | ● | ● | ● |
| primary_key | ● |  |  | ● |  |
| nullable | ● | ● | ● | ● |  |
| default | ● | ● | ● | ● | ● |

The implementation was to return the above.
However, max_length in SQLite3 always returns null.

Should the manual hastily describe the current situation described above?

**Comment by kenjis** [comment]
@naente  If we add the description, it is better to describe what's missing.

> max_length in SQLite3 always returns null.

It seems that max_length in SQLite3 does not make sense.
See https://www.sqlite.org/faq.html#q9

**Comment by naente** [comment]
@kenjis Thanks. I added a note about the currently supported databases.

**Comment by kenjis** [comment]
Closed by #6440
